ido-vertical-mode.el

```diff
@@ -107,38 +122,56 @@ so we can restore it when turning `ido-vertical-mode' off")
         (progn
           (setq additional-items-indicator "\n")
           (setq comps (append comps (make-list (- (1+ ido-max-prospects) lencomps) "")))))
-
+    (dotimes (i ido-max-prospects)
+      (setf (nth i comps) (substring (nth i comps) 0))
+      (when (string-match name (nth i comps))
+        (ignore-errors
+          (add-face-text-property (match-beginning 0)
+                                  (match-end 0)
+                                  'ido-vertical-match-face
+                                  nil (nth i comps)))))
     (if (and ind ido-use-faces)
         (put-text-property 0 1 'face 'ido-indicator ind))
-
     (if (and ido-use-faces comps)
         (let* ((fn (ido-name (car comps)))
                (ln (length fn)))
+          (setcar ido-vertical-decorations (format " [%d]\n-> " lencomps))
           (setq first (format "%s" fn))
```

Comment:
Sure, that's fine! Didn't want to step on your toes, but I can take care of the rest, whenever I get around to it :D

Thanks for contributing :+1: